packages/chevrotain/src/parse/cst/cst.ts

```diff
@@ -51,6 +53,64 @@ export function addNoneTerminalToCst(
     } else {
         node.children[ruleName].push(ruleResult)
     }
+
+    setNodeLocation(node, ruleResult.location)
+}
+
+function setNodeLocation(node: CstNode, locationInformation: any) {
```

Comment:
1. maybe too many IF conditions here to check the properties exist, after the first tume we assign location info all the properties should exist, so maybe we can only check if one property is missing?

2. Do not allow casting to boolean in the conditions,
 - ```javascript
       // implicit casting
       if (x) {}
       // equivalent to:
        if (x !== undefined || if x !== [] || if x !== "" || x !== 0 /* ... */ ) {}
       // no casting and faster performance veersion
       if  (x !== undefined) {}
   ``

Reply:
Ok for 2

For 1, I was wondering: is all information (endOffset, startLine, endLine, etc...) always present ? If no, should we always return some value (like NaN) if not present ?